### PR TITLE
Switch to promises

### PR DESCRIPTION
**Pull request by grabbou**
Uses `promise` aligned with react-native one, removes `async` dependency and `callback` for testing (we can now use `.then`)

### CODE REVIEW

File: src/link.js [hunk 1]
```diff
@@ -106,13 +102,12 @@ module.exports = function link(config, args, callback) {
...
-  const tasks = dependencies.map((dependency) => (next) =>
-    async.waterfall([
-      dependency.config.commands.prelink || commandStub,
-      makeLink(project, dependency),
-      dependency.config.commands.postlink || commandStub,
-    ], next)
-  );
+  const tasks = dependencies.map(dependency => () => {
+    const pre = dependency.config.commands.prelink || commandStub;
+    const post = dependency.config.commands.postlink || commandStub;
```
Comment (grabbou) [inline src/link.js]: @Kureev I think we need to update these in `rnpm` to return a promise and bump `minor`

File: src/link.js [hunk 2]
```diff
@@ -106,13 +106,12 @@ module.exports = function link(config, args, callback) {
...
-  const tasks = dependencies.map((dependency) => (next) =>
-    async.waterfall([
-      dependency.config.commands.prelink || commandStub,
-      makeLink(project, dependency),
-      dependency.config.commands.postlink || commandStub,
-    ], next)
-  );
+  const tasks = dependencies.map(dependency => () => {
+    const pre = promisify(dependency.config.commands.prelink || commandStub);
+    const post = promisify(dependency.config.commands.postlink || commandStub);
```
Comment (grabbou) [inline src/link.js]: Can we make these to return promise in `rnpm` ?

**Comment by grabbou** [comment]
@Kureev please check comments on outdated diffs since they are still relevant, just Github hiding them :)

**Comment by Kureev** [comment]
In general LGTM

**Comment by grabbou** [comment]
"In general" haha :D

**Pull request merged**